src/PlatformEngines.jl

```diff
@@ -214,21 +207,40 @@ function probe_platform_engines!(;verbose::Bool = false)
     # package_opts_functor, list_opts_functor, parse_functor).  The probulator
     # will check each of them by attempting to run `$test_cmd`, and if that
     # works, will set the global compression functions appropriately.
+
+    # the regex at the last position is meant for parsing the symlinks from verbose 7z-listing
+    # "Path = ([^\r\n]+)\r?\n" matches the symlink name which is followed by an optional return and a new line
+    # (?:[^\r\n]+\r?\n)+ = a group of non-empty lines (information belonging to one file is written as a block of lines followed by an empty line)
+    # more info on regex and a powerful online tester can be found at https://regex101.com
+    # Symbolic Link = ([^\r\n]+)"s) matches the source filename
     gen_7z = (p) -> (unpack_7z(p), package_7z(p), list_7z(p), parse_7z_list, r"Path = ([^\r\n]+)\r?\n(?:[^\r\n]+\r?\n)+Symbolic Link = ([^\r\n]+)"s)
     compression_engines = Tuple[]
 
-    write("demoTar.txt","Demo file for tar listing")
+    (tmpfile, io) = mktemp()
```

Comment:
More idiomatic:

```julia
mktemp() do (tmpfile, io)
	write(io, ...)
end
```

Also I imagine you want to get rid of the `@info(tmpfile)` below.

Reply:
This is also not a good place to use this, as we need to use `tmpfile` later.